lib/Service.php

```diff
@@ -189,6 +189,27 @@ function write($rootElementName, $value, $contextUri = null) {
 
     }
 
+    /**
+     * Maps the given plain-old-php-object (popo) to the given rootElementName within the given namespace.
+     *
+     * The $rootElement must be specified in clark notation.
+     *
+     * @param string $rootElementName
+     * @param object $popo
+     * @param string $namespace
+     */
+    function mapPopo($rootElementName, $popo, $namespace) {
+        if (!is_object($popo)) {
```

Comment:
It's a bit more of a design decision, because whatever we allow is what people will have to do... and elsewhere in the source generally rely on PHP class names (such as `$classMap` and `$elementMap`) so it would be internally consistent.

So between the two that would be my first choice. Not only for consistency, but it also plays a bit nicer with the idea that 'objects should be fully initialized after creation', and you can argue that a 'template object' is not a fully constructed object.